hstrat/_auxiliary_lib/_lazy_attach.py

```diff
@@ -0,0 +1,45 @@
+from typing import Callable, Optional, Any
+
+from lazy_loader import attach
+
+
+def lazy_attach(
+    __name: str,
+    *,
+    submodules: Optional[list[str]],
+    submod_attrs: Optional[dict[str, list[str]]],
+    launder: bool,
+    launder_names: Optional[list[str]] = None
+) -> tuple[Callable[[str], Any], Callable[[], list[str]], list[str]]:
+    """
```

Comment:
Should have a high level sentence explaining in general terms why/how this is used in the project.